lib/routes/profile.js

```diff
@@ -8,11 +8,19 @@ const checksum = require('checksum');
 
 const batch = require('../batch');
 
-function hasProfileScope(scopes) {
+const ALLOWED_SCOPES = {
+  'profile': true,
+  'profile:': true,
+  //openid connect
+  'email': true
+};
+
+function hasAllowedScope(scopes) {
+  console.log('hasAllowedScope: ', scopes);
   for (var i = 0, len = scopes.length; i < len; i++) {
-    var scope = scopes[i];
+    var scope = scopes[i].substr(0, 'profile:'.length);
```

Comment:
Neat trick, but it seems a little implicit and fragile to me, e.g. it depends on `'profile:'` being the longest entry in the `ALLOWED_SCOPES` set.  Given how important scope-matching is to security, I think we should prefer something simple and verbose over something short and clever.

(Of course, what we should really do is use a common helper lib for doing scope matching so that we only have to get it right once, but that's outside the scope of this bug).